src/search/view/SearchView.ts

```diff
@@ -588,6 +703,41 @@ export class SearchView extends BaseTopLevelView implements TopLevelView<SearchV
 		})
 	}
 
+	private renderCalendarTimeRangeField(): Children {
+		const startDate = new Date()
+		const start = this.searchViewModel.startDate == null ? lang.get("today_label") : formatDate(this.searchViewModel.startDate)
+		const endDate = incrementMonth(startDate, 2)
+		const end = this.searchViewModel.endDate == null ? formatDate(endDate) : formatDate(this.searchViewModel.endDate)
+		const timeDisplayValue = start + " - " + end
+		return m(TextField, {
+			label: "periodOfTime_label",
+			value: timeDisplayValue,
+			disabled: true,
+			class: "plr-button",
+			injectionsRight: () =>
+				m(IconButton, {
+					title: "selectPeriodOfTime_label",
+					click: async () => {
```

Comment:
I generally ask people to not make event handlers `async` and instead delegate to `async` functions